### DISCUSSION TITLE
Getting a Multiaddr not supported error

### DISCUSSION THREAD

**Discussion by harsh4786** [libp2p/rust-libp2p#4463] [Q&A] [answered]
Im trying to use the following as a multiaddress `Transport([("/dns4/wormhole-mainnet-v2-bootstrap.certus.one/udp/8999/quic", MultiaddrNotSupported("/dns4/wormhole-mainnet-v2-bootstrap.certus.one/udp/8999/quic"))])`

what could be the reasons a Multiaddr is not supported?

**Comment by thomaseizinger** [comment]
`/quic` refers to the `draft-29` version of QUIC which you need to explicitly enable in `libp2p::quic::Config`. Use `/quic-v1` instead!

**Reply by harsh4786** [reply]
As we discussed in this [issue](https://github.com/libp2p/rust-libp2p/discussions/4425). The nodes that im trying to connect to don't support `quic-v1` and instead support `quic`

**Reply by thomaseizinger** [reply]
Did you enable `draft-29` support in the config and it is still not working?

**Reply by harsh4786** [reply]
Yep enabled it and not working, it actually fails to dial successfully, it's a dialing error because `dial()` fails.

**Reply by thomaseizinger** [reply]
Can you share your code?

**Reply by harsh4786** [reply]
Yeah sure. https://gist.github.com/harsh4786/aa1693100a70c925c8ffd6f8ef55d070

**Reply by thomaseizinger** [reply, marked as answer]
Your issue is likely not QUIC but that fact that you are trying to dial `/dns` addresses without the DNS-transport :)

**Reply by harsh4786** [reply]
Damn, I honestly thought dns came baked in, in quic, are there any examples that do dns configuration?

**Reply by harsh4786** [reply]
Mdns should work though correct?

**Reply by thomaseizinger** [reply]
> are there any examples that do dns configuration?

A quick full-text search for "DNS" reveals that the `dcutr` example has some.

> Mdns should work though correct?

Yes, mDNS returns you IP addresses, not domains.

**Reply by harsh4786** [reply]
Thank you so much!

**Comment by thomaseizinger** [comment]
We have an open issue to improve this situation: https://github.com/libp2p/rust-libp2p/issues/2980

Let me know if you are interested in contributing!
